### ISSUE TITLE
Testing on the training set ? mAp result

### ISSUE THREAD

**Issue by thibo73800** [ChunML/ssd-tf2#11]
Hello,

I'm trying to test out your model. I have trained the ssd300 model for 100 epochs. Using your original test script. I get 0.65mAp and the following output :

aeroplane: 1.0
bicycle: 1.0
bird: 0.8
boat: 0.75
bottle: 0.36530612244897953
bus: 0.6666666666666666
car: 0.21311218901580342
cat: 0.2946236559139785
chair: 0.6521739130434783
cow: 1.0
diningtable: 1.0
dog: 0.6451612903225806
horse: 0.6923076923076923
motorbike: 0.4718614718614718
person: 0.08298857633796458
pottedplant: 0.24180327868852458
sheep: 0.6666666666666666
sofa: 0.8
train: 0.7777777777777778
tvmonitor: 1.0
mAP: 0.6560224650525793

But looking at the test script all the images are included rather than only the validation set. 
After changing the test script to select only the validation set I get the following. 

aeroplane: nan
bicycle: nan
bird: nan
boat: 0.125
bottle: nan
bus: nan
car: 0.22222222222222224
cat: nan
chair: 0.0
cow: nan
diningtable: nan
dog: 0.041666666666666664
horse: 0.16666666666666666
motorbike: 0.125
person: 0.03806079381701687
pottedplant: 0.047619047619047616
sheep: 0.375
sofa: nan
train: nan
tvmonitor: nan
mAP: nan

Even though there is an issue with the mAp computation, most of the class are not detected properly. Do you have any thoughts about the reason for this huge gap in performance? 

Thanks,

**Comment by ChunML** [comment]
That's weird to me. How many images did you run the test on?
